### PR TITLE
A block was not correctly recognize.

### PR DESCRIPTION
**Pull request by guy-maurel**
Add some more LOG_FMT for a better logging
A bugfix proposal for #1758

### CODE REVIEW

File: tests/config/bug_1758.cfg [hunk 1]
```diff
@@ -0,0 +1,2043 @@
+#
```
Comment (CDanU) [inline tests/config/bug_1758.cfg]: That needs a run through the reduction script.

I remember that you had a reason to postpone the usage of it but I don't quiet remember anymore why and if it is applicable for this case. I believe it was something related to the Unity fork.

File: tests/config/bug_1758.cfg [hunk 2]
```diff
@@ -0,0 +1,11 @@
+output_tab_size                 = 4
+sp_arith                        = force
+sp_assign                       = force
+sp_compare                      = force
+sp_before_sparen                = force
+indent_columns                  = 4
+nl_end_of_file                  = force
+nl_end_of_file_min              = 2
```
Comment (CDanU) [inline tests/config/bug_1758.cfg]: Still not to happy about the config, some of those options are definitely not needed. This needs a alteration of the test file. Id also like to see a test case where we have something like: `while(){if(){}}`

Comment (guy-maurel) [reply tests/config/bug_1758.cfg]: Why not "happy"?
This is the output of option_reducer.py

Comment (CDanU) [reply tests/config/bug_1758.cfg]: Yes the reduction script tries to reduce the option without changing the output formatting. The problem in this case is that the output file itself is to complex, it need to be simplified.

The initial problem is that the for braces are disappearing when they should not.

That means we definitely do not need any of:
```sql
sp_*
nl_end_of_file                  = force
nl_end_of_file_min              = 2
```

we most likely do not need:
```sql
output_tab_size                 = 4
indent_columns                  = 4
```

and  _maybe_ we do need:
```sql
nl_for_brace                    = remove
nl_after_if                     = force
mod_full_brace_if               = remove
```

**Comment by CDanU** [comment]
So now only the example with the on liner is missing:
> I'd also like to see a test case here where we have something like: for(){if(){}}

And it would be nice if you could reformat the test input with an empty config so that we can get a diff in the end that is only depending on the test options (excluding changes that are made by the default options)

**Comment by guy-maurel** [comment]
I prefer a new test case for your proposal.

File: src/braces.cpp [hunk 3]
```diff
@@ -528,14 +558,18 @@ static void examine_brace(chunk_t *bopen)
                || chunk_is_token(pc, CT_WHILE)
                || chunk_is_token(pc, CT_SWITCH)
                || chunk_is_token(pc, CT_USING_STMT)
-               || (  chunk_is_token(pc, CT_BRACE_OPEN)
-                  && chunk_is_token(prev, CT_FPAREN_CLOSE)))
+               || chunk_is_token(pc, CT_BRACE_OPEN)) // Issue #1758
```
Comment (mihaipopescu) [inline src/braces.cpp]: Really hard to tell apart changes. Is this the only functional change?

Comment (guy-maurel) [reply src/braces.cpp]: It is the consequence of the bug #1758 
The condition  `&& chunk_is_token(prev, CT_FPAREN_CLOSE)` was not apropriated to solve the case.

Comment (mihaipopescu) [reply src/braces.cpp]: The question is "Is this the **only** change that fixes the bug" ?

Comment (mihaipopescu) [reply src/braces.cpp]: because they are mixed with other changes, that's why I ask.

Comment (guy-maurel) [reply src/braces.cpp]: Yes. It is the only line.
I made lot of changes by the use of the debug function `LOG_FMT(...)`
to find out what the matter is.
I might be better to make to PR for that.

Comment (mihaipopescu) [reply src/braces.cpp]: no, it's fine like this but yeah, maybe next time different commits with different things changed would help in review.

Comment (guy-maurel) [reply src/braces.cpp]: Anybody to approve?

Comment (CDanU) [reply src/braces.cpp]: Hmm that whole function is a bit strange ... here is one failure:

```cpp
int main()
{

for(int f=0; f < 1; f++)
{
	auto a = int{1};
}

return 0;
}
```

**Comment by mihaipopescu** [comment]
@CDanU what do you think about the recent changes?

**Pull request merged**